README.md

```diff
@@ -274,6 +272,20 @@ Generate data for test cases matching a pytest pattern:
 CACHI2_GENERATE_TEST_DATA=true tox -e integration -- -k gomod
 ```
 
+### Adding new dependencies to the project
+
+Sometimes when working on adding a new feature you may need to add a new dependency to the project.
+Usually, one commonly goes about it by adding the dependency to one of the ``requirements`` files.
```

Comment:
Modern python projects should use pyproject.toml https://packaging.python.org/en/latest/guides/writing-pyproject-toml/#dependencies-and-requirements (we use it too - it would be good to mention it here)